dias/analyzers/source_spectra_analyzer.py

```diff
@@ -0,0 +1,733 @@
+                        start=int(
+                            np.argmin(np.abs(file_timestamp - source_start_time))
+                        ),
+                        stop=int(np.argmin(np.abs(file_timestamp - source_stop_time))),
+                        freq_sel=freq_sel,
+                        datasets=["vis", "flags/vis_weight"],
+                        apply_gain=False,
+                        renormalize=False,
+                    )
+
+                    bflag = (bdata.weight[:] > 0.0).astype(np.float32)
+                    bvar = tools.invert_no_zero(bdata.weight[:])
+
+                    lmbda = (
+                        scipy.constants.c * 1e-6 / bdata.freq[:, np.newaxis, np.newaxis]
+                    )
+
+                    # Loop over polarizations
+                    for ii, pol in enumerate(pols):
+
+                        self.logger.info("Processing Pol %s" % pol)
+
+                        pvis = bdata.vis[:, prod[pol], :]
+                        pvar = bvar[:, prod[pol], :]
+                        pflag = bflag[:, prod[pol], :]
+                        pcnt = cnt[np.newaxis, prod[pol], :]
+                        pscale = scale[pol][np.newaxis, :, np.newaxis]
+
+                        fringestop_phase = tools.fringestop_phase(
+                            ha,
+                            lat,
+                            src_dec,
+                            dist[pol][np.newaxis, :, 0, np.newaxis] / lmbda,
+                            dist[pol][np.newaxis, :, 1, np.newaxis] / lmbda,
+                        )
+
+                        vis[freq_sel, ii, :] += np.sum(
+                            pscale * pcnt * pflag * pvis * fringestop_phase, axis=1
+                        )
+                        var[freq_sel, ii, :] += np.sum(
+                            (pscale * pcnt) ** 2 * pflag * pvar, axis=1
+                        )
+                        counter[freq_sel, ii, :] += np.sum(
+                            pscale * pcnt * pflag, axis=1
+                        )
+
+                    self.logger.info(
+                        "Took %0.1f seconds to process this block."
+                        % (time.time() - t0,)
+                    )
+
+                    del bdata
+                    gc.collect()
+
+                # Normalize
+                inv_counter = tools.invert_no_zero(counter)
+                vis *= inv_counter
+                var *= inv_counter ** 2
+
+                ra = np.degrees(np.unwrap(ra))
+                ha = ra - np.degrees(src_ra)
+
+                # Fit response to model
+                if self.perform_fit:
+
+                    fwhm = np.zeros((nfreq, npol), dtype=np.float32)
+                    for ii in range(npol):
+                        fwhm[:, ii] = cal_utils.guess_fwhm(
+                            data.freq, pol=pols[ii][0], dec=src_dec, sigma=True
+                        )
+
+                    flag = counter > 0.0
+
```

Comment:
There are tools for performing these types of fits in `ch_util.cal_utils` that you might consider using instead of replicating that code in the ancillary functions below.  For example:

```
fitter = cal_utils.FitGaussAmpPolyPhase(poly_deg_phi=self.poly_deg_phi)
fitter.fit(ha, vis, np.sqrt(var), width=fwhm, absolute_sigma=True)
resid = vis - fitter.predict(ha)
resid_rms = np.std(resid, axis=-1)
```
Then `poly_deg_phi` would need to be added as a config property and should probably default to 3.